examples/riot/hello_lf/Makefile

```diff
@@ -1,14 +1,20 @@
-# name of your application
-LF_MAIN = HelloLF
+REACTOR_UC_PATH ?= $(CURDIR)/../../../
 
-# This has to be the absolute path to the RIOT base directory:
-# TODO: In the lingo template this should point to ./RIOT because we plan to include RIOT as a submodule there
-RIOTBASE ?= $(CURDIR)/../../../../RIOT
+# The name of the LF application inside "./src" to build/run/flash etc.
+LF_MAIN ?= HelloLF
+
+# Enable reactor-uc features
+# CFLAGS += -DNETWORK_CHANNEL_TCP_POSIX
+# CFLAGS += -DNETWORK_CHANNEL_COAP_RIOT
 
-# Path of generated lf c-code
-LF_SRC_GEN_PATH = $(CURDIR)/src-gen/$(LF_MAIN)
+# Execute the LF compiler if build target is "all"
+ifeq ($(MAKECMDGOALS),all)
+  _ :=  $(shell $(REACTOR_UC_PATH)/lfc/bin/lfc-dev src/$(LF_MAIN).lf)
```

Comment:
Just for my own understanding, what does this syntax here achieve? Will the command be invoked before any target is run?

Reply:
This is a small hack, it copies the output of the shell script over to the variable in this case underscore. This way the evaluation of the makefile is blocked until the script is finished ;)

Reply:
That's very clever!